src/cmake/modules/TargetTools.cmake

```diff
@@ -1617,11 +1617,13 @@ function (basis_finalize_targets)
     basis_make_target_uid (TARGET_UID_basis_sh basis_sh)
     basis_make_target_uid (TARGET_UID_basis_py basis_py)
     basis_make_target_uid (TARGET_UID_Basis_pm Basis_pm)
-    list (REMOVE_ITEM TARGETS
-      ${TARGET_UID_basis_sh}
-      ${TARGET_UID_basis_py}
-      ${TARGET_UID_Basis_pm}
-    )
+    if (TARGETS)
```

Comment:
Thanks, @wittenbe, I always forget that CMake throws an error when the lists are empty. I think CMake should just do not perform any such list operation in this case... possibly just a matter of motivating a discussion on their users list.

Thanks, for the fix. I think it would be better, however, to directly return from the function when the TARGETS list returned by the `basis_get_target_property` call in the else block is empty. This saves needless extra work and avoids any further potential for code assuming the list to be non-empty.

Do you mind adding the following right after [this line](https://github.com/wittenbe/BASIS/blob/14d5d288825881e271146e650402df929a504679/src/cmake/modules/TargetTools.cmake#L1613):
```cmake
if (NOT TARGETS)
  return ()
endif ()
```
instead of the ```if (TARGETS)``` block around the `list` command?

I.e., revert this last commit using `git reset`, make the suggested modification, and then commit this change instead to your master branch.

P.S.: For future PR's, it is best to use a separate feature/bugfix branch in your fork so you can still amend the changes before they are merged and make possibly other unrelated modifications/fixes in different branches.